packages/runtime/container-runtime/src/test/gc/garbageCollection.spec.ts

```diff
@@ -1740,4 +1663,226 @@ describe("Garbage Collection Tests", () => {
+				assert.deepStrictEqual(
+					gcStats,
+					expectedStats,
+					"The stats for first GC run should be same as initial stats",
+				);
+
+				// Unreference another data store node.
+				defaultGCData.gcNodes[nodes[0]] = [];
+
+				// There should be 3 unreferenced node / data stores.
+				// There should be 1 node / data store whose reference state got updated.
+				expectedStats.unrefNodeCount = 3;
+				expectedStats.unrefDataStoreCount = 3;
+				expectedStats.updatedNodeCount = 1;
+				expectedStats.updatedDataStoreCount = 1;
+
+				gcStats = await garbageCollector.collectGarbage({});
+				assert.deepStrictEqual(gcStats, expectedStats, "Incorrect GC stats 1");
+
+				// Add a new reference.
+				defaultGCData.gcNodes[nodes[0]] = [nodes[2]];
+
+				// There should be 2 unreferenced node / data stores now.
+				// There should be 1 node / data store whose reference state got updated.
+				expectedStats.unrefNodeCount = 2;
+				expectedStats.unrefDataStoreCount = 2;
+				expectedStats.updatedNodeCount = 1;
+				expectedStats.updatedDataStoreCount = 1;
+
+				gcStats = await garbageCollector.collectGarbage({});
+				assert.deepStrictEqual(gcStats, expectedStats, "Incorrect GC stats 2");
+			});
+
+			it("can generate stats with deleted nodes", async () => {
+				const expectedStats = initialStats;
+				let gcStats = await garbageCollector.collectGarbage({});
+				assert.deepStrictEqual(
+					gcStats,
+					expectedStats,
+					"The stats for first GC run should be same as initial stats",
+				);
+
+				// Advance the clock past sweep timeout so that unreferenced nodes are deleted.
+				clock.tick(sweepTimeoutMs + 1);
+
+				// There should be 2 deleted nodes and data stores. There shouldn't be any nodes whose
+				// reference state updated.
+				expectedStats.deletedNodeCount = 2;
+				expectedStats.deletedDataStoreCount = 2;
+				expectedStats.updatedNodeCount = 0;
+				expectedStats.updatedDataStoreCount = 0;
+
+				gcStats = await garbageCollector.collectGarbage({});
+				assert.deepStrictEqual(gcStats, expectedStats, "Incorrect GC stats");
+			});
+
+			it("can generate stats with deleted nodes after multiple sweep runs", async () => {
+				const expectedStats = initialStats;
+				let gcStats = await garbageCollector.collectGarbage({});
+				assert.deepStrictEqual(
+					gcStats,
+					expectedStats,
+					"The stats for first GC run should be same as initial stats",
+				);
+
+				// Advance the clock past sweep timeout so that unreferenced nodes are deleted.
+				clock.tick(sweepTimeoutMs + 1);
+
+				// There should be 2 deleted nodes and data stores. There shouldn't be any nodes whose
+				// reference state updated.
+				expectedStats.deletedNodeCount = 2;
+				expectedStats.deletedDataStoreCount = 2;
+				expectedStats.updatedNodeCount = 0;
+				expectedStats.updatedDataStoreCount = 0;
+
+				gcStats = await garbageCollector.collectGarbage({});
+				assert.deepStrictEqual(gcStats, expectedStats, "Incorrect GC stats 1");
```

Comment:
Why does this work when Sweep is disabled?

Reply:
We use sweep ready nodes as "deleted". I added a comment to the delete tests to highlight this.